src/cider/tests/nextgen/functional/CastTest.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <gtest/gtest.h>
+#include "tests/utils/CiderTestBase.h"
+
+#define GEN_CAST_TYPE_TEST_CLASS_FOR_ARROW(C_TYPE_NAME, SUBSTRAIT_TYPE_NAME)             \
+  class Cast##C_TYPE_NAME##TypeTestForArrow : public CiderTestBase {                     \
+   public:                                                                               \
+    Cast##C_TYPE_NAME##TypeTestForArrow() {                                              \
+      table_name_ = "test";                                                              \
+      create_ddl_ = "CREATE TABLE test(col_a " #C_TYPE_NAME ", col_b " #C_TYPE_NAME ")"; \
+      QueryArrowDataGenerator::generateBatchByTypes(                                     \
+          schema_,                                                                       \
+          array_,                                                                        \
+          20,                                                                            \
+          {"col_a", "col_b"},                                                            \
+          {CREATE_SUBSTRAIT_TYPE(SUBSTRAIT_TYPE_NAME),                                   \
+           CREATE_SUBSTRAIT_TYPE(SUBSTRAIT_TYPE_NAME)},                                  \
+          {0, 2},                                                                        \
+          GeneratePattern::Random,                                                       \
+          -100,                                                                          \
```

Comment:
Do we need to test some failed cases like overflow?

Reply:
you mean primitive type cast check? indeed we need support, I will fix cast overflow check and add related tests after runtime error_code is ready.https://github.com/intel/BDTK/pull/280